File: 2025/day3/part1.cpp
#include<iostream>
#include<fstream>
#include<string>
using namespace std;
int main(){
    ifstream file;
    file.open("input.txt");
    if(!file.is_open()){
        cout<<"Error opening file\n";
        return 1;
    }
    string line;
    int sum =0;
    while(getline(file,line)){
        char max1='0',max2='0';
        int j;
        int n=line.length();
        for(int i=0;i<n-1;i++){
            if(max1<line[i]){
                max1=line[i];
                j=i;
            }
        }
        for(int i=j+1;i<n;i++){
            if(max2<line[i]){
                max2=line[i];
            }
        }
        cout<<max1<<" and "<<max2<<endl;
        int num = ((max1-'0')*10)+(max2-'0');
        //cout<<num<<endl;
        sum+=num;
        //cout<<sum<<endl;
    }
    cout<<sum;
    return 0;
}

File: 2025/day4/part1.py
grid = []
with open("input.txt") as f:
    for line in f:
        grid.append(list(line.strip()))

r = len(grid)
c = len(grid[0])
count=0
for i in range(0,r):
    for j in range(0,c):
        p=0
        if grid[i][j]=='@':
            for k in range(i-1,i+2):
                if k < 0 or k>r-1 :
                    continue
                for l in range(j-1,j+2):
                    if l< 0 or l>c-1 :
                        continue
                    if grid[k][l]=="@":
                        p=p+1
            if p<=4:
                count+=1
print (count)